File: agent/0815/onboarding-agent/mobile/android/OnboardingAgentGating.kt
package ai.onboarding

object LocalGateCfg {
    const val LOW = 0.45
    const val HIGH = 0.75
}

class OnDeviceLLM {
    fun inferIntent(text: String): Triple<String, Double, Map<String, String>> {
        val label = if (text.contains("오류") || text.contains("에러")) "troubleshooting" else "faq"
        val score = if (text.contains("사진") || text.contains("이미지")) 0.7 else 0.5
        return Triple(label, score, emptyMap())
    }
}

class Gating {
    private val llm = OnDeviceLLM()
    fun route(text: String): Action {
        val (label, score, slots) = llm.inferIntent(text)
        return when {
            score < LocalGateCfg.LOW -> Action.Clarify("질문을 조금 더 구체적으로 말씀해 주세요")
            score < LocalGateCfg.HIGH -> Action.Escalate("/chat", mapOf("text" to text, "intent" to label, "slots" to slots))
            else -> Action.LocalAnswer("간단 답변: $label")
        }
    }
}

sealed class Action {
    data class Clarify(val msg: String): Action()
    data class Escalate(val path: String, val payload: Map<String, Any?>): Action()
    data class LocalAnswer(val msg: String): Action()
}

File: agent/0815/onboarding-agent/mobile/ios/OnboardingAgentGating.swift
import Foundation

struct LocalGateCfg { static let low = 0.45; static let high = 0.75 }

class OnDeviceLLM {
  func inferIntent(_ text: String) -> (String, Double, [String:String]) {
    let label = (text.contains("오류") || text.contains("에러")) ? "troubleshooting" : "faq"
    let score = (text.contains("사진") || text.contains("이미지")) ? 0.7 : 0.5
    return (label, score, [:])
  }
}

enum Action {
  case clarify(String)
  case escalate(String, [String:Any])
  case localAnswer(String)
}

class Gating {
  let llm = OnDeviceLLM()
  func route(_ text: String) -> Action {
    let (label, score, slots) = llm.inferIntent(text)
    if score < LocalGateCfg.low { return .clarify("질문을 조금 더 구체적으로 말씀해 주세요") }
    if score < LocalGateCfg.high { return .escalate("/chat", ["text": text, "intent": label, "slots": slots]) }
    return .localAnswer("간단 답변: \(label)")
  }
}
